src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java

```diff
@@ -149,6 +149,33 @@ public void should_not_attach_exception_record_with_pending_payments_when_classi
         );
     }
 
+    @Test
+    void should_set_exceptionRecordReference_in_scanned_documents() throws Exception {
+        //given
+        CaseDetails caseDetails = ccdCaseCreator.createCase(emptyList(), Instant.now());
+        CaseDetails exceptionRecord = createExceptionRecord("envelopes/supplementary-evidence-envelope.json");
+
+        // when
+        attachExceptionRecord(caseDetails, exceptionRecord, null);
+
+        //then
+        await("Exception record is attached to the case")
+            .atMost(60, TimeUnit.SECONDS)
+            .pollDelay(2, TimeUnit.SECONDS)
+            .until(() -> isExceptionRecordAttachedToTheCase(caseDetails, 1));
+
+        CaseDetails updatedCase = ccdApi.getCase(
+            String.valueOf(caseDetails.getId()),
+            caseDetails.getJurisdiction()
+        );
+
+        List<ScannedDocument> scannedDocuments = getScannedDocuments(updatedCase);
+        assertThat(scannedDocuments).hasSize(1);
+        for (ScannedDocument scannedDocument: scannedDocuments) {
```

Comment:
don't need a for loop you know from previous step that only one scanned doc is present